### ISSUE TITLE
jupace autoboot_command wont type "GR" but "gr" works

### ISSUE THREAD

**Issue by mcfilmmakers** [mamedev/mame#7690]
I have been trying to get autoboot commands to work in Jupiter Ace and came across an interesting bug:

autoboot_command "load GR\n" results in typing load R and enter instead of the expected load GR and enter.

however, autoboot_command "load gr\n" outputs the correct load gr and enter.

The issue is that the GR is case sensitive for some, but not all games. Example: Ace invaders (Forth Dimension) aka invader1

That's the big issue.

Secondly, the platofrm is heavily reliant on typing a command, play the tape a bit, pause the tape, type 2nd command, play more tape, etc, etc.... is there a way of doing this in mame?

Third, how does one type a character that requires holding shift? ex: shift+2 = @ instead of 2.

**Comment by cuavas** [comment]
You might need to use an autoboot Lua script if it’s any more complex than just waiting a certain amount of time and then typing something.

**Comment by mcfilmmakers** [comment]
In regards to my 2nd and 3rd issues, you're probably right. However the MAME wiki doesn't explain how to do this at all. I would very much appreciate a working example of such a script.

That said, the main issue is that autoboot_command should be typing the characters I tell it to type, and it isn't.

**Comment by Bob-Z** [comment]
For what is worth I have written some autoboot LUA scripts here: https://github.com/Bob-Z/RandoMame/tree/main/autoboot_script

They are "just working" level scripts, I am LUA newbie and they are certainly not the best LUA code you can imagine. But they mostly work (to be honest, I have not tested all of them with MAME 227, which changed the LUA API, so they may not all work actually...)

Anyway, I hope this helps. And, if you manage to write a working script for jupace, I will be thankfull if you can share it.

**Comment by angelosa** [comment]
```lua
emu.keypost('abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ')
```
![0000](https://user-images.githubusercontent.com/9251271/107149086-03b1fe80-6957-11eb-8fa0-222dbc98136f.png)

Most likely down to a jupace driver issue rather than LUA, cfr. `mame x1turbo40 hubasic1 -console`

![0024](https://user-images.githubusercontent.com/9251271/107149231-bd10d400-6957-11eb-871d-e313d9b605e1.png)

**Comment by cuavas** [comment]
Yeah, the issue is that the fixed keypress duration used by the natural keyboard manager isn’t enough for the Jupiter Ace.  Natural keyboard mode is decidedly flaky.

**Comment by mcfilmmakers** [comment]
Except that the issue is unique to those missing characters and not the keyboard as a whole.
